odl/solvers/smooth/nonlinear_cg.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2014-2017 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Nonlinear version of conjugate gradient."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+
+from odl.solvers.util import ConstantLineSearch
+
+
+__all__ = ('conjugate_gradient_nonlinear',)
+
+
+def conjugate_gradient_nonlinear(f, x, line_search=1.0, maxiter=1000, nreset=0,
+                                 tol=1e-16, beta_method='FR',
+                                 callback=None):
+    """Conjugate gradient for nonlinear problems.
+
+    Notes
```

Comment:
Ok, but in the long run I find it preferable to get directly to the parameters without much ado since that's what users have to look at most frequently. The math explanation is useful once or a couple of times, but later it's just in the way to the parameter reference.
So I prefer the note section after Parameters in almost all cases.